.travis.yml

```diff
@@ -7,6 +7,23 @@ matrix:
   include:
   - jdk: oraclejdk7
     script: mvn -q deploy --settings .travis-settings.xml -Dno.gem.deploy=true
+  - jdk: oraclejdk7
+    env: ANDROID_EXAMPLES=true
+    language: android
+    android:
+      components:
+        - build-tools-21.1.1
+        - platform-tools
+        - extra-android-support
+        - android-21
+        - sys-img-armeabi-v7a-android-21
+    script: mvn install -P android-examples -pl examples/android,examples/android/android-test -am -amd -Dno.gem.deploy=true -Dandroid.device=test
+    install: mvn install -P android-examples -pl examples/android,examples/android/android-test -am -amd -DskipTests=true -Dmaven.javadoc.skip=true -B -V
```

Comment:
I did not know that travis defaults to that, however I think this is a highly inefficient way of implementing a build pipeline and from the maven point of view actually extremely weird as well.

I wonder why travis would do that, considering they actually pay for more execution time.

Anyway so you are saying that if we would to leave out the install step travis would run it anyway but instead build the whole project since it does not use the correct profiles and locations? If that's the case, then I would recommend skipping the install step explicitly like that ```install: true```. I am so focused on performance, because when we work with emulators we should try to improve every bit of the build pipeline.

Using ```mvn verify``` instead of ```mvn install``` also saves some time by avoiding unnecessary maven steps. I agree that ```clean``` is unnecessary.